### PR TITLE
Remove old Dev health-data-assessment route logic and tests

### PR DESCRIPTION
**Pull request by alan-amida**
## What was the problem?

The endpoint /v1/health-data-assessment is a legacy end point and should be removed. It will be never called from [va.gov](http://va.gov/) so it is safe to remove it. v1/full-heath-data-assessment is what replaced it.

Associated tickets or Slack threads:

- [MCP-2057](https://amida.atlassian.net/browse/MCP-2057)

## How does this fix it?[^1]

- [x] Remove all artifacts related top v1/health-data-assessment. These include camel routes, test, endpoints, etc.
- [ ] ~~Create a /v2/health-data-assesment which has the same functionality as v1/full-heath-data-assessment. Rename the artifacts without “Full” in the name.~~
- [x] v1/full-health-data-assessment should stay functionally as is

## How to test this PR
- Step 1
- Step 2[^secrel]

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Review by au2 — DISMISSED** [review]
I would also remove PrimaryRoutes.ENDPOINT_SUBMIT_CLAIM in the PrimaryRoutes.java. I do not think it will used anymore.

**Comment by au2** [comment]
Also please add a brief description to branch names going forward. It helps me to find when I am reviewing multiple prs.

**Comment by au2** [comment]
One more thing. OpenApiConfiguration.java has some Swagger configuration. You want to remove Health Assessment from there. I wanted to move those hard coded section names to application.properties but never got there.

File: controller/src/main/java/gov/va/vro/controller/exception/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public ResponseEntity<ClaimProcessingError> handleClaimProcessingException(
   @ExceptionHandler(HttpMessageNotReadableException.class)
   public ResponseEntity<ClaimProcessingError> handleJsonParseException(
       JsonParseException exception) {
-    log.error("Unexpected error", exception);
+    log.error("Bad Request: Malformed JSON", exception);
```
Comment (alan-amida) [inline controller/src/main/java/gov/va/vro/controller/exception/GlobalExceptionHandler.java:50]: Adding better description (from last ticket)

**Comment by alan-amida** [comment]
@au2 did you still want me to ignore:

- [ ] Create a /v2/health-data-assesment which has the same functionality as v1/full-heath-data-assessment. Rename the artifacts without “Full” in the name.

for this ticket, or did you want me to go ahead and try to implement this here?

**Comment by alan-amida** [comment]
Also, thinking about this more, I have reservations about issuing a `201 Created` response for something, even if we're POSTing it, that isn't found. I feel like the proper response here should be `404 Not Found` -- but this also assumes a standardized REST endpoint (where we would have something like a `/v1/veteran/${icn}` or something) where it feels way too late in the game to discuss.

**Review by au2 — APPROVED** [review]
Looks like claim-submit is being used in multiple places. We will need to cleanup. Please write thw ticket.

**Review by ianwolson — APPROVED** [review]

**Pull request merged**